From 6cb9c79a1cd4140c34b0e05fb8a19f01fbd572c4 Mon Sep 17 00:00:00 2001
From: Shadichy <shadichy@blisslabs.org>
Date: Wed, 16 Apr 2025 01:06:02 +0700
Subject: [PATCH] Enable msse2 on i386 linux

Signed-off-by: Shadichy <shadichy@blisslabs.org>
---
 CMakeLists.txt | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 26fad80ab..faa8e5bcf 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -378,6 +378,10 @@ if(FIPS_DELOCATE OR NOT OPENSSL_NO_ASM)
     endif()
     if(NOT WIN32)
       set(CMAKE_ASM_FLAGS "${CMAKE_ASM_FLAGS} -Wa,--noexecstack")
+			if(CMAKE_SIZEOF_VOID_P EQUAL 4)
+				set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -msse2")
+				set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -msse2")
+			endif()
     endif()
     # Clang's integerated assembler does not support debug symbols.
     if(NOT CMAKE_ASM_COMPILER_ID MATCHES "Clang")
-- 
2.49.0

